core/domain/question_services_test.py

```diff
@@ -590,6 +590,105 @@ def test_get_interaction_id_for_question(self):
 
 class QuestionMigrationTests(test_utils.GenericTestBase):
 
+    def test_migrate_question_state_from_v30_to_v31(self):
+        answer_group = {
+            'outcome': {
+                'dest': 'abc',
+                'feedback': {
+                    'content_id': 'feedback_1',
+                    'html': '<p>Feedback</p>'
+                },
+                'labelled_as_correct': True,
+                'param_changes': [],
+                'refresher_exploration_id': None,
+                'missing_prerequisite_skill_id': None
+            },
+            'rule_specs': [{
+                'inputs': {
+                    'x': 'Test'
+                },
+                'rule_type': 'Contains'
+            }],
+            'training_data': [],
+            'tagged_skill_misconception_id': None
+        }
+        question_state_dict = {
+            'content': {
+                'content_id': 'content_1',
+                'html': 'Question 1'
+            },
+            'recorded_voiceovers': {
+                'voiceovers_mapping': {}
+            },
+            'written_translations': {
+                'translations_mapping': {
+                    'explanation': {}
+                }
+            },
+            'interaction': {
+                'answer_groups': [answer_group],
+                'confirmed_unclassified_answers': [],
+                'customization_args': {
+                    'buttonText': {
+                        'value': 'Add item'
+                    }
+                },
+                'default_outcome': {
+                    'dest': None,
+                    'feedback': {
+                        'content_id': 'feedback_1',
+                        'html': 'Correct Answer'
+                    },
+                    'param_changes': [],
+                    'refresher_exploration_id': None,
+                    'labelled_as_correct': True,
+                    'missing_prerequisite_skill_id': None
+                },
+                'hints': [{
+                    'hint_content': {
+                        'content_id': 'hint_1',
+                        'html': 'Hint 1'
+                    }
+                }],
+                'solution': {
+                    'correct_answer': 'This is the correct answer',
+                    'answer_is_exclusive': False,
+                    'explanation': {
+                        'content_id': 'explanation_1',
+                        'html': 'Solution explanation'
+                    }
+                },
+                'id': 'TextInput'
```

Comment:
Hi @iamprayush, this might be why L232-234 of question_domain isn't covered. Can you try changing this to SetInput?

Reply:
@kevintab95 Thanks for the suggestion! This, along with some other minor changes is now covering those lines.